src/components/useRedirect.ts

```diff
@@ -0,0 +1,24 @@
+import { useCallback } from 'react';
+import { RouteComponentProps } from 'react-router';
+
+interface Config {
+  url: string;
+  keys?: string[];
+  cookie?: string;
+}
+
+const useRedirect = (history: RouteComponentProps['history'], config: Config) => {
+  return useCallback(() => {
+    const { url, keys = [], cookie } = config;
+
+    for (const key of keys) {
+      window?.sessionStorage?.removeItem(key);
```

Comment:
Why does this need optional chaining?

Reply:
This is the original code:
`
if (window && window.sessionStorage) {
  window.sessionStorage.removeItem('token');
}
`
That's why I added optional chaining